cpufeatures/src/aarch64.rs

```diff
@@ -0,0 +1,112 @@
+//! ARM64 CPU feature detection support.
+//!
+//! Unfortunately ARM instructions to detect CPU features cannot be called from
+//! unprivileged userspace code, so this implementation relies on OS-specific
+//! APIs for feature detection.
+
+#[macro_export]
+#[doc(hidden)]
+macro_rules! __unless_target_features {
+    ($($tf:tt),+ => $body:expr ) => {
```

Comment:
Played around with a few different syntaxes for this macro and this is the best one I could come up with.

Open to suggestions for alternatives.